Source/RunActivity/Viewer3D/Popups/HUDWindow.cs

```diff
@@ -474,6 +474,10 @@ void TextPageCommon(TableData table)
             TableAddLabelValue(table, Viewer.Catalog.GetString("FPS"), "{0:F0}", Viewer.RenderProcess.FrameRate.SmoothedValue);
             TableAddLine(table);
 
+            // Used in axle module to adjust substeps for frame rate drops
+            var mstsLocomotive = Viewer.PlayerLocomotive as MSTSLocomotive;
+            Viewer.PlayerTrain.ScreenFrameRate = Viewer.RenderProcess.FrameRate.SmoothedValue;
```

Comment:
You maybe correct, but I am not sure how to call the base value that would be maintained up to date. 

Do you have any suggestions.

Reply:
A good place may be MSTSLocomotiveViewer.PrepareFrame

Reply:
I am not certain as to how to render RenderProcess.FrameRate.SmoothedValue visible in MSTSLocomotiveViewer.PrepareFrame, so it will need to stay as it is unless I can get some code as an example of making it visible.

Reply:
But this only works if the HUD is open, which is not ideal. Why don't you use elapsedClockSeconds, which is directly obtained at the Axle.Update function, and smooth that yourself if required?

Reply:
I would prefer to use the FrameRate values, as this is the parameter which is being impacted, otherwise we would need to find the correct correlation between the FrameRate and various values of elapsedClockSeconds.

I suspect that most people will have the HuD open (even if it is only the main HuD), so this should be ok.

I would like to have this approved now and then we will see what other "negative" or "positive" feedback that we get in the next stage.

As a background task, I will see if I can find a way to read the FrameRate "continuously".

Reply:
> I suspect that most people will have the HuD open (even if it is only the main HuD), so this should be ok.
That's a huge assumption, and I don't think it is true. Anyway, limiting adhesion to work only when the HUD is opened is a mistake. I'm sorry, but I cannot approve it until this is fixed. We don't have to be in a hurry to merge PR which contains serious defects as this one, but we should take some time to fix them instead of including bad code into OR.

> I would prefer to use the FrameRate values, as this is the parameter which is being impacted, otherwise we would need to find the correct correlation between the FrameRate and various values of elapsedClockSeconds.
Your code is basically saying: "if frame rate is low, I do not care about you, I'll just give lots of numerical errors to force you to switch to simple adhesion" I don't feel like this is the right approach, but since there is no agreement on this, I won't make further comments about it.

Re elapsedClockSeconds and FPS, when simulation time is at 100%, FPS=1/elapsedClockSeconds, so you could possibly use that.